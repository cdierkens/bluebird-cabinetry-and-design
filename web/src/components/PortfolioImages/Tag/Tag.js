import PropTypes from "prop-types";
import React from "react";

const className =
  "cursor-pointer inline-block rounded p-3 m-1 sm:m-2 focus:outline-none focus:ring shadow-md";

const Tag = ({ label, onChange, value, checked }) => {
  return (
    <label
      className={`${className} ${
        checked ? "bg-blue-dark text-white" : "bg-white text-blue-dark"
      }`}
    >
      <input
        aria-label={label}
        className="sr-only"
        type="checkbox"
        onChange={onChange}
        value={value}
        checked={checked}
      />
      <span>{label}</span>
    </label>
  );
};

Tag.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  checked: PropTypes.bool,
};

export default Tag;
