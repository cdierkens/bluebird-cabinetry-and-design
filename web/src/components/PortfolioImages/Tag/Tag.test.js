import { render } from "@testing-library/react";
import React from "react";
import Tag from "./Tag";

it("matches snapshot", () => {
  const { asFragment } = render(<Tag label="mock label" />);

  expect(asFragment()).toMatchSnapshot();
});
