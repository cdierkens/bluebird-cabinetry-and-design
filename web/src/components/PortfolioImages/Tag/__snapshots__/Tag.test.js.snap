// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <label
    class="cursor-pointer inline-block rounded p-3 m-1 sm:m-2 focus:outline-none focus:shadow-outline shadow-md bg-white text-blue-dark"
  >
    <input
      class="sr-only"
      type="checkbox"
      value=""
    />
    <span>
      mock label
    </span>
  </label>
</DocumentFragment>
`;
